File: app/src/main/java/com/security/passwordmanager/TimePickerActivity.java
package com.security.passwordmanager;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TimePicker;

import androidx.activity.result.contract.ActivityResultContract;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.appcompat.app.AppCompatActivity;

import com.security.passwordmanager.settings.Support;

import java.util.Calendar;

public class TimePickerActivity extends AppCompatActivity {

    private static final String ARG_TIME = "time";
    private static final String ARG_TITLE = "title";
    private static final String EXTRA_TIME = "extra_time";

    private TimePicker mTimePicker;

    @NonNull
    public static Intent newIntent(Context context, Calendar calendar, @StringRes int title) {
        Intent intent = new Intent(context, TimePickerActivity.class);
        intent.putExtra(ARG_TIME, calendar);
        intent.putExtra(ARG_TITLE, title);

        return intent;
    }

    public static Calendar parseIntent(Intent intent) {
        return (Calendar) intent.getSerializableExtra(EXTRA_TIME);
    }

    @SuppressLint("InflateParams")
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Support support = Support.getInstance(this);

        setContentView(R.layout.dialog_time);

        Calendar calendar = (Calendar) getIntent().getSerializableExtra(ARG_TIME);
        int title = getIntent().getIntExtra(ARG_TITLE, R.string.start_time);
        setTitle(title);

        int hours = calendar.get(Calendar.HOUR_OF_DAY);
        int minutes = calendar.get(Calendar.MINUTE);

        mTimePicker = (TimePicker) getLayoutInflater().inflate(
                support.isLightTheme() ? R.layout.time_picker_light : R.layout.time_picker_dark,
                null
        );
        mTimePicker.setIs24HourView(true);
        mTimePicker.setHour(hours);
        mTimePicker.setMinute(minutes);

        ((LinearLayout) findViewById(R.id.dialog_time_picker)).addView(mTimePicker);

        Button buttonCancel = findViewById(R.id.dialog_button_cancel);
        Button buttonOk = findViewById(R.id.dialog_button_ok);

        findViewById(R.id.constraint_layout_buttons).setBackgroundColor(support.getBackgroundColor());

        buttonCancel.setOnClickListener(v -> {
            setResult(RESULT_CANCELED);
            finish();
        });

        buttonOk.setOnClickListener(v -> {

            calendar.set(Calendar.HOUR_OF_DAY, mTimePicker.getHour());
            calendar.set(Calendar.MINUTE, mTimePicker.getMinute());

            Intent intent = new Intent();
            intent.putExtra(EXTRA_TIME, calendar);
            setResult(RESULT_OK, intent);
            finish();
        });
    }

//    @NonNull
//    @Override
//    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
//
//
//        return new AlertDialog.Builder(requireActivity())
//                .setView(v)
//                .setTitle(title)
//                .setPositiveButton(android.R.string.ok, (dialog, which) -> {
//
//                    Date date1 = new GregorianCalendar(
//                            calendar.get(Calendar.YEAR),
//                            calendar.get(Calendar.MONTH),
//                            calendar.get(Calendar.DAY_OF_MONTH),
//                            mTimePicker.getHour(),
//                            mTimePicker.getMinute()
//                    ).getTime();
//
//                    sendResult(date1);
//                })
//                .create();
//    }

    public static class TimePickerActivityContract extends ActivityResultContract<Calendar, Calendar> {

        @StringRes
        private final int title;

        public TimePickerActivityContract(@StringRes int title) {
            this.title = title;
        }

        @NonNull
        @Override
        public Intent createIntent(@NonNull Context context, Calendar calendar) {
            return newIntent(context, calendar, title);
        }

        @Override
        public Calendar parseResult(int resultCode, @Nullable Intent intent) {
            if (resultCode != Activity.RESULT_OK || intent == null)
                return null;

            return parseIntent(intent);
        }
    }
}

File: app/src/main/java/com/security/passwordmanager/TimePickerActivity.kt
package com.security.passwordmanager

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.TimePicker
import androidx.activity.result.contract.ActivityResultContract
import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatActivity
import com.security.passwordmanager.databinding.DialogTimeBinding
import com.security.passwordmanager.viewmodel.SettingsViewModel
import java.util.*

class TimePickerActivity : AppCompatActivity() {

    companion object {
        private const val ARG_TIME = "time"
        private const val ARG_TITLE = "title"
        private const val EXTRA_TIME = "extra_time"

        fun newIntent(context: Context, calendar: Calendar, @StringRes title: Int) =
            createIntent<TimePickerActivity>(context) {
                putExtra(ARG_TIME, calendar)
                putExtra(ARG_TITLE, title)
            }

        fun parseIntent(intent: Intent) =
            intent.getSerializableExtra(EXTRA_TIME) as Calendar



        class TimePickerActivityContract(@StringRes private val title: Int) :
            ActivityResultContract<Calendar, Calendar?>() {

            override fun createIntent(context: Context, input: Calendar) =
                newIntent(context, input, title)

            override fun parseResult(resultCode: Int, intent: Intent?) = when {
                resultCode != RESULT_OK || intent == null -> null
                else -> parseIntent(intent)
            }
        }
    }

    private lateinit var binding: DialogTimeBinding
    private lateinit var timePicker: TimePicker

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DialogTimeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val settings = SettingsViewModel.getInstance(this)
        val calendar = intent.getSerializableExtra(ARG_TIME) as Calendar
        val title = intent.getIntExtra(ARG_TITLE, R.string.start_time)

        setTitle(title)

        val hours = calendar[Calendar.HOUR_OF_DAY]
        val minutes = calendar[Calendar.MINUTE]

        timePicker = layoutInflater.inflate(
            if (settings.isLightTheme) R.layout.time_picker_light
            else R.layout.time_picker_dark,
            null,
            false
        ) as TimePicker

        timePicker.setIs24HourView(true)
        timePicker.hour = hours
        timePicker.minute = minutes

        binding.dialogTimePicker.addView(timePicker)

        binding.constraintLayoutButtons.root.setBackgroundColor(settings.backgroundColor)
        settings.updateThemeInScreen(window, supportActionBar)


        binding.constraintLayoutButtons.dialogButtonCancel.setOnClickListener {
            setResult(RESULT_CANCELED)
            finish()
        }

        binding.constraintLayoutButtons.dialogButtonOk.setOnClickListener {
            calendar[Calendar.HOUR_OF_DAY] = timePicker.hour
            calendar[Calendar.MINUTE] = timePicker.minute

            val intent = Intent()
            intent.putExtra(EXTRA_TIME, calendar)
            setResult(RESULT_OK, intent)
            finish()
        }
    }
}